features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Shared/AppsAndKeys/TokenManager.jsx

```diff
@@ -129,6 +129,33 @@ class TokenManager extends React.Component {
         this.loadApplication();
     }
 
+
+    /**
+     * get supported grant types from the settings api
+     */
+    getserverSupportedGrantTypes = () => {
+        const api = new API();
+        const promisedSettings = api.getSettings();
+        promisedSettings
+            .then((response) => {
+                let { keyRequest } = this.state;
+                keyRequest = { ...keyRequest };
```

Comment:
Shall we do like below?

```javascript
const keyRequest  = { ...this.state.keyRequest} ;
```

Reply:
fixed in https://github.com/wso2/carbon-apimgt/pull/7260/commits/d6515e0ab107d623b98febb7e2a336a5434c775e